app/src/main/kotlin/com/wire/android/ui/calling/OngoingCallViewModel.kt

```diff
@@ -32,108 +18,34 @@ import javax.inject.Inject
 class OngoingCallViewModel @Inject constructor(
     savedStateHandle: SavedStateHandle,
     private val navigationManager: NavigationManager,
-    private val conversationDetails: ObserveConversationDetailsUseCase,
-    private val allCalls: GetAllCallsUseCase,
-    private val endCall: EndCallUseCase,
-    private val muteCall: MuteCallUseCase,
-    private val unMuteCall: UnMuteCallUseCase,
-    private val setVideoPreview: SetVideoPreviewUseCase,
-    private val updateVideoState: UpdateVideoStateUseCase
+    private val ongoingCall: GetOngoingCallUseCase
 ) : ViewModel() {
 
-    var callEstablishedState by mutableStateOf(OngoingCallState())
-
     val conversationId: QualifiedID = savedStateHandle
         .get<String>(EXTRA_CONVERSATION_ID)!!
         .parseIntoQualifiedID()
 
     init {
         viewModelScope.launch {
-            launch { initializeScreenState() }
-            launch { observeOngoingCall() }
-        }
-    }
-
-    private suspend fun initializeScreenState() {
-        conversationDetails(conversationId = conversationId)
-            .collect {
-                callEstablishedState = when (it) {
-                    is ConversationDetails.Group -> {
-                        callEstablishedState.copy(
-                            conversationName = it.conversation.name
-                        )
-                    }
-                    is ConversationDetails.OneOne -> {
-                        callEstablishedState.copy(
-                            conversationName = it.otherUser.name,
-                            avatarAssetId = it.otherUser.completePicture?.let { assetId -> UserAvatarAsset(assetId) }
-                        )
-                    }
-                    is ConversationDetails.Self -> throw IllegalStateException("Invalid conversation type")
-                }
+            val job = launch {
+                ongoingCall().first { it.isNotEmpty() }
             }
-    }
-
-    private suspend fun observeOngoingCall() {
-        allCalls().collect {
-            if (it.first().conversationId == conversationId)
-                when (it.first().status) {
-                    CallStatus.CLOSED -> navigateBack()
-                    else -> {
-                        print("DO NOTHING")
-                    }
-                }
-        }
-    }
-
-    fun hangUpCall() {
-        viewModelScope.launch {
-            endCall(conversationId)
-            navigateBack()
-        }
-    }
-
-    fun setVideoPreview(view: View?) {
-        viewModelScope.launch {
-            setVideoPreview(conversationId, PlatformView(view))
+            job.join()
+            // We start observing once we have an ongoing call
+            observeCurrentCall()
```

Comment:
not 100 % sure but instead of job.join() , we can achieve the same iwth : 

viewmoDelScope.launch{
ongoingCall().first { it.isNotEmpty() }
observeCurrentCall()
}

Reply:
right